src/pages/index.js

```diff
@@ -0,0 +1,291 @@
+  setButtonText(submitButton, true);
+  api
+    .editAvatar({
+      avatar: avatarLinkInput.value,
+    })
+    .then((data) => {
+      avatarImage.src = data.avatar;
+      closeModal(avatarModal);
+      evt.target.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
+}
+
+function handleCardFormSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+  api
+    .addNewCard({ name: cardNameInput.value, link: cardLinkInput.value })
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      disableButton(cardSubmitButton, settings);
+      closeModal(addCardModal);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
+}
+
+let selectedCard;
+let selectedCardId;
+
+function getCardElement(data) {
+  console.log(data);
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardName = cardElement.querySelector(".card__text");
+  const cardImage = cardElement.querySelector(".card__image");
+  const heartButton = cardElement.querySelector(".card__heart-button");
+  const trashButton = cardElement.querySelector(".card__trash-button");
+
+  cardName.textContent = data.name;
+  cardImage.src = data.link;
+  cardImage.alt = data.name;
+
+  if (data.isLiked) {
+    heartButton.classList.toggle("card__heart-button_liked");
+  }
+  heartButton.addEventListener("click", (evt) => {
+    handleLike(evt, data._id);
+  });
+  trashButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data)
+  );
+  cardImage.addEventListener("click", () => handleImageClick(data));
+
+  return cardElement;
+}
+
+function handleLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__heart-button_liked");
+  api
+    .changeLikeStatus(id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__heart-button_liked");
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleImageClick(data) {
+  openModal(modalPreview);
+  modalImageName.textContent = data.name;
+  modalImage.src = data.link;
+  modalImage.alt = data.name;
+}
+
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(modalDelete);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true, "Deleting...", "Delete");
+  api
+    .removeCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(modalDelete);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(setButtonText(submitButton, false, "Deleting...", "Delete"));
```

Comment:
`    .finally(setButtonText(submitButton, false, "Deleting...", "Delete"));
`

In `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 


```js

    .finally(() => {
      setButtonText(submitButton, false, "Deleting...", "Delete")
    });

```